velox/exec/Aggregate.h

```diff
@@ -193,6 +202,13 @@ class Aggregate {
   // 'groups'. No-op for fixed length accumulators.
   virtual void destroy(folly::Range<char**> /*groups*/) {}
 
+  // Returns the type that extractAccumulator will produce. This is
+  // different from resultType_ for complex final aggregations. Needed
+  // for e.g. spilling.
+  virtual TypePtr accumulatorType() const {
```

Comment:
This API seems redundant. We already know signatures of all the aggregate functions and can use SignatureBinder to compute intermediate and final types given raw input types. See resolveAggregateType in PlanBuilder.cpp.